// audioconcat/index.js
var merge = require('lodash.merge')
var ffmpeg = require('fluent-ffmpeg')
var version = require('./package.json').version

module.exports = exports = function (inputs, opts) {
  return new Audioconcat(inputs, opts)
}
console.log('#1')
exports.VERSION = version
console.log('#2')
exports.ffmpeg = ffmpeg
console.log('#3')


function Audioconcat(inputs, opts) {
  console.log('#4')
  this.inputs = inputs || []
  this.opts = opts || {}
}

console.log('#5')
Audioconcat.prototype.options = function (opts) {
  console.log('#6')
  merge(this.opts, opts)
  return this
}
console.log('#7')
Audioconcat.prototype.concat = function (file) {
  console.log('#8')
  if (file) {
    this.opts.output = file
  }
  return concat(this.inputs, this.opts)
}
console.log('#9')

function concat(inputs, opts) {
  console.log('#10')
  var filter = 'concat:' + inputs.join('|')
  console.log('#11')
  var renderer = ffmpeg()
    .input(filter)
    .outputOptions('-acodec copy')
    console.log('#12')
  var output = opts.output
  if (output) {
    return renderer.save(output)
  }
  console.log('#13')
  return renderer
}
console.log('#14')
